src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/ImporterService.cs

```diff
@@ -100,6 +109,37 @@ public class ImporterService : IImporterService
             GeographicLevel.PlanningArea
         };
 
+        private static readonly List<GeographicLevel> AllowedSoloGeographicLevels = new List<GeographicLevel>
+        {
+            GeographicLevel.Provider,
+            GeographicLevel.School,
+        };
+
+        public static bool HasSoloAllowedGeographicLevel(HashSet<GeographicLevel> subjectGeographicLevels)
+        {
+            if (subjectGeographicLevels.Count == 1)
+            {
+                foreach (var allowedSoloLevel in AllowedSoloGeographicLevels)
+                {
+                    if (subjectGeographicLevels.Contains(allowedSoloLevel))
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        public static bool AllowRowImport(HashSet<GeographicLevel> subjectGeographicLevels, GeographicLevel rowGeographicLevel)
```

Comment:
I think the methods `AllowRowImport` and `HasSoloAllowedGeographicLevel` and probably the fields `IgnoredGeographicLevels` and `AllowedSoloGeographicLevels` don't really belong in ImporterService becaused they are used by `ValidatorService` and `SplitFileService` as well as `ImporterService`. I think you should consider abstracting them out to a common util class, which could have class or method level comments explaining what all this is about and why we do it. It's not particularly obvious at the moment that we have this logic around ignoring school/provider in certain circumstances and I think that would be a good place to add the comments.

Reply:
I've moved them into ImporterUtils and added comments to HasSoloAllowedGeographicLevel and AllowRowImport. Tried to balance the comment text between saying what they do while not have the comments easily go stale. Happy to change the comments to something else if you'd prefer.